src/devtools/client/debugger/src/components/SecondaryPanes/ObjectPreviewSuspenseCacheAdapter.tsx

```diff
@@ -2,16 +2,31 @@ import { preCacheObjects } from "@bvaughn/src/suspense/ObjectPreviews";
 import { trackExecutionPointPauseIds } from "@bvaughn/src/suspense/PauseCache";
 import { ExecutionPoint, PauseData, PauseId } from "@replayio/protocol";
 import { addPauseDataListener, removePauseDataListener } from "protocol/thread/pause";
-import { useEffect } from "react";
+import { useLayoutEffect } from "react";
 import { useFeature } from "ui/hooks/settings";
 
+function createReplacer() {
+  const cache = new Set();
+  return function replacer(_: string, value: any): any {
+    if (typeof value === "object" && value !== null) {
+      if (cache.has(value)) {
+        return undefined;
+      }
+      cache.add(value);
+    }
+    return value;
+  };
+}
```

Comment:
We need to **deeply** clone data because of `ValueFront` mutations (😔) so `JSON.stringify` can be helpful. Our data _might_ have circular references though, so we need this replacer.

**TODO** Verify if this is still the case or if it was only because of the `ValueFront` mutations.

Reply:
I think this is safe to remove now actually.